cpp/modmesh/simd/simd.hpp

```diff
@@ -39,56 +39,13 @@ namespace modmesh
 namespace simd
 {
 
-namespace detail
-{
-#ifndef NDEBUG
-template <typename T>
-bool is_aligned(T const * pointer, size_t alignment)
-{
-    return (reinterpret_cast<std::uintptr_t>(pointer) % alignment) == 0;
-}
-
-template <typename T>
-void check_alignment(T const * pointer, size_t required_alignment, const char * name)
-{
-    if (!is_aligned(pointer, required_alignment))
-    {
-        std::fprintf(stderr,
-                     "Warning: %s pointer %p is not aligned to %zu bytes. "
-                     "SIMD performance may be degraded.\n",
-                     name,
-                     static_cast<const void *>(pointer),
-                     required_alignment);
-    }
-}
-#endif
-
-// Get the recommended memory alignment for SIMD operations based on the detected SIMD instruction set.
-inline constexpr size_t get_recommended_alignment()
-{
-#if defined(__aarch64__) || defined(__arm__)
-    return 16;
-#elif defined(__AVX512F__)
-    return 64;
-#elif defined(__AVX__) || defined(__AVX2__)
-    return 32;
-#elif defined(__SSE__) || defined(__SSE2__) || defined(__SSE3__) || defined(__SSSE3__) || defined(__SSE4_1__) || defined(__SSE4_2__)
-    return 16;
-#else
-    return 0;
-#endif
-}
-
-} // namespace detail
-
 // Check if each element from start to end (excluded end) is within the range [min_val, max_val)
 template <typename T>
 const T * check_between(T const * start, T const * end, T const & min_val, T const & max_val)
 {
-    using namespace detail;
-    switch (detect_simd())
+    switch (detail::detect_simd())
```

Comment:
We should not `using namespace detail`, or defeat part of the purpose of the namespace `detail`.  I created issue #631 to get rid of the misusing.